### PR TITLE
More informative page titles

### LINKED ISSUE #9822: [RFE] More specific page titles
**Is your feature request related to a problem? Please describe.**
Right now the page title for rancher is always Rancher. If you want to go back to a previous page in your browser history, this makes it hard to find the right page to go back to.

**Describe the solution you'd like**
It would be nice if the page title was more specific, like:
 - `Rancher - <cluster name> - Deployments`
 - `Rancher - <cluster name> - Deployment: phpmyadmin`
 - `Rancher - CD - <git repo>`
 - `Rancher - VM mgmt - <harvester cluster name> - Hosts`
 - `Rancher - VM mgmt - <harvester cluster name> - Image: ubuntu-22.04-minimal-cloudimg-amd64`
 - etc.

**Describe alternatives you've considered**
An alternative is that Rancher keeps track of a page history itself, but that seems clunky and the browser doesn't provide the functionality without a reason.

**Additional context**
afbeelding

### PR DESCRIPTION
**Pull request by codyrancher**
### Summary
Fixes #9822

### Screenshot/Video
image

### CODE REVIEW

**Comment by codyrancher** [comment]
@aalves08 I wanted to make a note about the approach because I still think it's a little odd.

I decided to make use of `<TabTitle>` instead of using another mechanism for two reasons:
- There are places that the titles are created purely in the template and not in the js at all
- Most places the title isn't known until translations are loaded which can cause the title to update which then means each page would need to watch for changes in order to render things properly.

I justified it as being okay because typically one sets the title on the page using the `<title>` element.



One other option I thought of that I think could lead to similar results but doesn't address the first point above would be to create something like:
```
computed: {
   ...computedTitle('propToWatch', options)
}
```

**Review by aalves08 — DISMISSED** [review]
I've been trying to wrap my head around this since my initial thought was always for this logic to be bound to routes (in the sense that we could assess the route the user is in and set the title) but because of translations I don't think that's possible... 🤔 Can we chat a bit about that?

Nevertheless, the logic you implemented seems correct and it's generally working. Just a couple of items to point out:

- There seems to be a problem assessing the title for the "entry" route of `Kubewarden`. Isn't it possible to get the `product` and add it to the title? I understand that the view title might a problem

Screenshot 2024-02-02 at 11 01 39
 
 - We should add a unit test for the `TabTitle` component. We this we can test the component directly and, indirectly, the `title` for the app

**Review by jordojordo — COMMENTED** [review]
Overall it looks good to me, the concept of using a component seems appropriate considering the amount of variations we have throughout the code-base which determines a page title.

Is it possible to override the breadcrumbs altogether? Also is it possible to add icons/svg's into the title? Perhaps both of these are out of the scope of this PR, but it could be some nice improvements.

File: shell/components/nav/Type.vue [hunk 1]
```diff
@@ -178,7 +185,7 @@ export default {
       @mouseenter="setNear(true)"
       @mouseleave="setNear(false)"
     >
-      <span class="label">{{ type.label }}&nbsp;<i class="icon icon-external-link" /></span>
+      <span class="label">{{ type.label }}&nbsp;<i class="icon icon-external-link" />Hello</span>
```
[the remark this answers is not in the document]
Comment (codyrancher) [reply shell/components/nav/Type.vue:200]: Definitely not lol. Removed.

**Comment by codyrancher** [comment]
> Is it possible to override the breadcrumbs altogether? 

It's possible to override the breadcrumbs. You can see that in the `should only show child` unit test.

> Also is it possible to add icons/svg's into the title? Perhaps both of these are out of the scope of this PR, but it could be some nice improvements.

I don't think that this is possible. I think you're wanting to change the tab favorite icon which isn't a part of the title.

**Comment by codyrancher** [comment]
> We should add a unit test for the TabTitle component. We this we can test the component directly and, indirectly, the title for the app

I've added unit and e2e tests. As long as nothing is flaky it should be ready for you to look at tomorrow. I'll rerun or fix tests if anything fails.

**Review by jordojordo — APPROVED** [review]

**Pull request merged**